### PR TITLE
vdk-core: Support Python 3.10

### PR DESCRIPTION
**Pull request by gageorgiev**
This change makes it so vdk-core supports Python 3.10.
Addresses #390 

Signed-off-by: gageorgiev <gageorgiev@vmware.com>

### CODE REVIEW

**Comment by tozka** [comment]
Don't forget to mark PR as draft (https://github.blog/2019-02-14-introducing-draft-pull-requests/) 

Also, please split into 2 commits or 2 PRs for vdk-control-cli and vdk-core. 

If you go with 2 commits remember to use to rebase option when merging .

File: projects/vdk-core/.gitlab-ci.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ image: "python:3.9"
   stage: build
   before_script:
     - cd projects/vdk-core/
+    - pip install -U pip
+    - pip install ./plugins/vdk-test-utils # TODO: REMOVE THIS AT SOME POINT
```
Comment (tozka) [inline projects/vdk-core/.gitlab-ci.yml:22]: why is that necessary? build.sh does the same `pip install -e plugins/vdk-test-utils`

Comment (gageorgiev) [reply projects/vdk-core/.gitlab-ci.yml]: ditched it, ditched the line under it as well as it was a duplicate too

**Review by YanaZhivkova — APPROVED** [review]
Looks good to me

**Review by tozka — APPROVED** [review]
